cfgov/tccp/jinja2/tccp/includes/apr_rating.html

```diff
@@ -0,0 +1,23 @@
+{%- macro apr_rating(rating, count=none) -%}
+    {%- set label = apr_rating_lookup[rating] -%}
+
+    <div class="m-apr-rating m-apr-rating--{{ label }}">
+        <dt>
```

Comment:
@contolini @niqjohnson SonarCloud is complaining about the lack of enclosing `<dl>` here. Thoughts?

<img width="545" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/e43704e2-8d9b-4cb9-89a1-f9344a998a0c">

Reply:
@chosak, at first I thought this was just SonarCloud not understanding that this is an include and the `dt` will be in a `dl` when it's rendered. But now I see we have a couple of variations of the markup, and it will only render in a `dl` in one of them. `dt`s and `dd`s do indeed have to be in a `dl`, so this markup will be invalid in the results cards and the APR tables.

I'm not sure what the best fix is, though. We could make the Jinja a little more complex and have it conditionally render as `dt` and `dd` sometimes and as a `div` other times (like in [the original list view macro](https://github.com/cfpb/consumerfinance.gov/pull/8413/files#diff-b1df41179c4cf86b41072300e0e5a260ca44823d2f29bbafb645ff91e1defa29L67-L77)). But I'm not sure we're getting enough benefit out of the `dt`/`dd` markup to make that complexity worth it. It's nice semantically, but since the only content in the `dt` right now is the SVG, I'm not sure we'll get anything out of that markup practically speaking. I'm leaning towards changing [the legend](https://github.com/cfpb/consumerfinance.gov/pull/8413/files#diff-b1df41179c4cf86b41072300e0e5a260ca44823d2f29bbafb645ff91e1defa29L43-L58) to no longer use a `dl` to simplify everything, but that's just a gut feeling. Any preferences, @contolini (or @anselmbradford)?

Reply:
Removing it makes sense. Although it is nice to have it say how many items are in the legend, which having it in a list does. You could also explore making the places its used a list (of one item) as well, and then refactoring the template that's shared to always have a list of one to three items. Or if you wanted to remove the list, but still want to add some semantics, you could maybe look into the use of the `<dfn>` element. However, this gets tricky because `<dfn>` needs to contain text. To get around this, you could set the title of the `<dfn>` to a unicode `$`, `$$`, and `$$$`, and then wrap both the icon and the accompanying text in the `<dfn>`:

```
<div><dfn title="$"><svg … ><span>Pay less interest (107 cards)</span></dfn></div>
```

Which would read _"dollar, definition"_, and then _"Pay less interest (107 cards)"_, and finally _"end of dollar, definition"_.

Reply:
> since the only content in the dt right now is the SVG

This is a great point because having a definition term that consists solely of SVGs is pretty useless to non-visual visitors and is related to our DS discussion about marking all our icons as [decorative](https://github.com/cfpb/design-system/issues/1965). If/when we hide our icons from assistive technologies we'll need to add some descriptive text for our APR ratings.

Re: the lack of a `dl`, we have an `o-apr-rating-group` element that is used for the global key but not individual card ratings. The easiest solution would be to apply that syntax to the card ratings while also putting some assistive text in the `dt` so that it's no longer just SVGs. Something like: https://github.com/cfpb/consumerfinance.gov/pull/8431/